my-app/src/context/functions/addToCart.js

```diff
@@ -1,13 +1,12 @@
 import React from 'react'
 import axios from "axios"
 
-const addToCart = ({id,title,author,price},encodedToken,dispatch) => {
-    //console.log("From add to cart",{'_id' : id , 'title' : title , 'author' : author , 'price' : price})
+const addToCart = ({id,title,author,price,category,rating},encodedToken,dispatch) => {
+   
     axios
-     .post("/api/user/cart",{"product" : { "_id" : id , "title" : title , "author" : author , "price" : price}},{headers :{ 'authorization' : encodedToken}})
+     .post("/api/user/cart",{"product" : { "_id" : id , "title" : title , "author" : author , "price" : price , "category" : category , "rating" : rating}},{headers :{ 'authorization' : encodedToken}})
```

Comment:
In places where the key and values are same ,no need to do "title":title . 
You can simply do {"_id" : id ,  title ,  author ,  price , category , rating}